### PR TITLE
Added Poisson Loss as an objective

### PR DESCRIPTION
**Pull request by anayebi**
I added Poisson loss as an objective. The Poisson loss is the negative log likelihood of data y_true given predictions y_pred, according to a Poisson model (assumes that y_pred > 0). Thus, the loss function is given by y_pred - y_true * log(y_pred).

This loss is typically used in computational neuroscience, when predicting the probability that a neuron will fire. Should be used after a softplus activation, to ensure that y_pred > 0. Often more useful in computational models of cells than the usual cross entropy loss as Poisson loss more heavily penalizes the model for predicting a spike during long periods of cell inactivity, than cross entropy would.

This complements PR https://github.com/fchollet/keras/pull/422.

### CODE REVIEW

**Comment by fchollet** [comment]
This PR is two lines in total, but it contains 14 commits. This is problematic as it makes the project history hard to read. Please rebase from master.

File: keras/objectives.py [hunk 1]
```diff
@@ -32,8 +32,8 @@ def squared_hinge(y_true, y_pred):
 
 def hinge(y_true, y_pred):
     return T.maximum(1. - y_true * y_pred, 0.).mean(axis=-1)
-
-
+    
+
```
Comment (fchollet) [inline keras/objectives.py]: Also, please no style changes, such as deletion of empty lines or insertion of tabs.

**Comment by anayebi** [comment]
Ok, fixed.

**Comment by fchollet** [comment]
Awesome, thanks.
